pancancer-arch-3/src/main/java/info/pancancer/arch3/reportbot/SlackReportBot.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2015 CancerCollaboratory
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package info.pancancer.arch3.reportbot;
+
+import com.ullink.slack.simpleslackapi.SlackSession;
+import com.ullink.slack.simpleslackapi.events.SlackMessagePosted;
+import com.ullink.slack.simpleslackapi.impl.SlackSessionFactory;
+import com.ullink.slack.simpleslackapi.listeners.SlackMessagePostedListener;
+import info.pancancer.arch3.Base;
+import info.pancancer.arch3.persistence.PostgreSQL;
+import info.pancancer.arch3.reporting.ReportAPI;
+import info.pancancer.arch3.reporting.ReportAPIFactory;
+import info.pancancer.arch3.reporting.SlackRenderer;
+import info.pancancer.arch3.utils.Constants;
+import info.pancancer.arch3.utils.Utilities;
+import java.io.IOException;
+import java.util.Locale;
+import org.apache.commons.configuration.HierarchicalINIConfiguration;
+
+/**
+ * This responds to Slack events and translates them into reporting calls.
+ *
+ * @author dyuen
+ */
+public class SlackReportBot extends Base {
+
+    public static final int SLEEP_IN_MILLISECONDS = 5000;
+
+    public static void main(String[] argv) throws Exception {
+        SlackReportBot bot = new SlackReportBot(argv);
+        bot.doWork();
+    }
+
+    public SlackReportBot(String[] argv) throws IOException {
+        super();
+        parseOptions(argv);
+    }
+
+    public void doWork() throws Exception {
+
+        final HierarchicalINIConfiguration settings = Utilities.parseConfig(configFile);
+        final PostgreSQL db = new PostgreSQL(settings);
+
+        final SlackSession session = SlackSessionFactory.createWebSocketSlackSession(settings.getString(Constants.REPORT_TOKEN));
+        session.addMessagePostedListener(new SlackMessagePostedListener() {
+            @Override
+            public void onEvent(SlackMessagePosted event, SlackSession session) {
+                System.out.println(event.toString());
```

Comment:
`println` calls should probably be replaced with using a proper logging framework.
But...
One problem with keeping all of these components (Worker, JobGenerator, Coordinator, and now Reporting) in one big fat jar file is that there's *one* logging config file for all the components so they all log to the same place (yes, we could probably create multiple loggers for each distinct function of the application, but it becomes a problem when a utility class method is invoked: which log file should *that* go to?). It's probably not a problem for Worker and JobGenerator/Coordinator (since Worker functionality should not be used on the same machine as is also using Generator and Coordinator), but since Generator, Coordinator and Reporter might all be run on the same system, having separate log files might be nice.